src/js/crypto.js

```diff
@@ -13,7 +13,74 @@
  * limitations under the License.
  */
 
-var hashes = ['sha1'];
+var shaTypes = {
+  'sha1': 4,
+  'sha256': 6,
+};
+
+var hashes = ['sha1', 'sha256'];
+
+function Verify(signtype) {
+  if (!(this instanceof Verify)) {
+    return new Verify(signtype);
+  }
+
+  signtype = signtype.toLowerCase();
+
+  if (hashes.indexOf(signtype) < 0) {
+    throw new Error('Unknown signing algorithm.' +
+                    'Please use crypto.getSignatures()');
+  }
+
+  Object.defineProperty(this, 'hashtype', {
+    value: signtype,
+    writable: false,
+    enumerable: true,
+  });
+}
+
+Verify.prototype.update = function(data) {
+  if (this.data) {
+    if (Buffer.isBuffer(data)) {
+      this.data = Buffer.concat([this.data, data]);
+      return;
+    }
+
+    this.data = Buffer.concat([this.data, new Buffer(data)]);
+    return;
+  }
+
+  if (Buffer.isBuffer(data)) {
+    this.data = data;
+    return;
+  }
+
+  this.data = new Buffer(data);
+};
+
+Verify.prototype.verify = function(publicKey, signature) {
+  if (this.data) {
+    var type = getHashType(this.hashtype);
+    var hash = native.shaEncode(this.data, type);
+    return native.rsaVerify(type, hash, publicKey, signature);
+  }
+
+  throw new Error('verify shouldn\'t be called on an empty Verify');
+};
+
+function getHashType(hashStr) {
+  var type;
+  if (hashStr.substring(0, 3) == 'sha') {
+    type = shaTypes.sha1;
+    switch (hashStr.substring(3, 6)) {
+      case '256': {
+        type = shaTypes.sha256;
+        break;
+      }
+    }
+  }
+  return type;
+}
```

Comment:
Is this function called only once? We have an array of types and this function, would be good to merge these into one, and call it max once for each Hash object.

Reply:
That is a good idea, and there is no possibility it would return null/undefined since creating the `verify` or `hash` object would return an error if there's no such algorithm.